crates/notion-core/src/tool.rs

```diff
@@ -348,3 +355,94 @@ impl Tool for Yarn {
         }
     }
 }
+
+impl Tool for Npm {
+    fn new(session: &mut Session) -> Fallible<Self> {
+        session.add_event_start(ActivityKind::Npm);
+
+        let mut args = args_os();
+        let exe = arg0(&mut args)?;
+        if let Some(ref platform) = session.current_platform()? {
+            session.prepare_image(platform)?;
+
+            if platform.yarn.is_some() {
+                // If Yarn is enabled for the current project, warn the user that they shouldn't be using npm
+                style::display_warning(&String::from("This project is configured to use yarn, npm should be avoided"));
```

Comment:
I'm pretty wary of adding warnings to Notion. Warnings are a risk in a tool that gets used regularly, because when users learn to ignore them, they depreciate the value of all console output.

Also, there may be some unintended consequences here. For example, you may be in a Yarn-based project but run a third-party script that does something benign with npm, like use it to query information about the package's version, and then that script will suddenly start generating this warning.